### PR TITLE
Timer rush

### PR DESCRIPTION
**Pull request by kojsmn**
We implemented the timer to keep track of time left in the game, so that way we can ensure we won't time out.

### CODE REVIEW

File: build/config/tournament.txt [hunk 1]
```diff
@@ -5,7 +5,7 @@ NUMCLIENTS=2
 NUMGAMES=1
 
 PORT=32101
-HOST=127.0.0.1
```
Comment (NigoroJr) [inline build/config/tournament.txt:8]: Keep this localhost.

File: src/Evaluators.h [hunk 2]
```diff
@@ -6,6 +6,8 @@
 #include <functional>
 #include <utility>
 
+#include <iostream>
```
Comment (NigoroJr) [inline src/Evaluators.h]: Not using.

File: src/Moderator.cpp [hunk 3]
```diff
@@ -69,19 +69,23 @@ unsigned Moderator::get_search_depth(const DomineeringState& state) const {
     unsigned game_moves = state.getNumMoves();
     unsigned depth;
 
+    if (searcher.get_time_left() > 20.0) {
```
Comment (NigoroJr) [inline src/Moderator.cpp]: Magic number?

File: src/Searcher.cpp [hunk 4]
```diff
@@ -1,10 +1,15 @@
 #include "Searcher.h"
+#include "GamePlayer.h"
+
+#include <iostream>
```
Comment (NigoroJr) [inline src/Searcher.cpp]: Not using.

File: src/Searcher.cpp [hunk 5]
```diff
@@ -31,8 +38,9 @@ Searcher& Searcher::operator=(const Searcher& other) {
     best_moves = other.best_moves;
     ordered_moves = other.ordered_moves;
     tp_table = other.tp_table;
+    timer = other.timer;
 
-    return *this;
+        return *this;
```
Comment (NigoroJr) [inline src/Searcher.cpp]: Indentation.

File: src/Searcher.cpp [hunk 6]
```diff
@@ -40,8 +48,9 @@ Searcher& Searcher::operator=(Searcher&& other) {
     best_moves = std::move(other.best_moves);
     ordered_moves = std::move(other.ordered_moves);
     tp_table = std::move(other.tp_table);
+    timer = std::move(other.timer);
 
-    return *this;
+        return *this;
```
Comment (NigoroJr) [inline src/Searcher.cpp]: Indentation.

File: src/Searcher.cpp [hunk 7]
```diff
@@ -64,7 +80,9 @@ Node Searcher::search(const DomineeringState& state,
     search_under(root, ab, state, depth_limit);
     move_thread = std::thread(&Searcher::move_order, this, root.team);
 
-    return best_moves.front();
+    timer.click();
+
+        return best_moves.front();
```
Comment (NigoroJr) [inline src/Searcher.cpp]: Indentation.

**Comment by NigoroJr** [comment]
Also, delete the binary file `src/.Timer.h.swo`.

File: src/Moderator.cpp (hunk 3 above)
Comment (kojsmn) [inline src/Moderator.cpp]: We kind of just chose that number. If you have a better number go for it!

-----Original Message-----

Comment (NigoroJr) [inline src/Moderator.cpp]: The value looks legitimate. Just create a `static constexpr float TIME_LIMIT = 20;` or something. Arbitrary number (other than 42) appearing in code almost never looks good.

Comment (kojsmn) [inline src/Moderator.cpp]: I pushed those changes! Why is 42 okay?

-----Original Message-----

**Comment by NigoroJr** [comment]
LGTM. Thanks :star2: 

42 is okay because it's "the answer to life the universe and everything."

**Comment by kojsmn** [comment]
Hahah that makes so much sense. So maybe we should make the time limit 42... Haha jk

-----Original Message-----

**Pull request merged**